### ISSUE TITLE
allow descriptions in scalafix:ok

### ISSUE THREAD

**Issue by fommil** [scalacenter/scalafix#702]
I'd like to allow user-provided descriptions of why a rule is overridden. e.g.

```
    URLEncoder.encode(s, "UTF-8") // scalafix:ok UTF-8 always succeeds
```

but this makes the suppression be ignored

**Comment by marcelocenerine** [comment]
yeah, this is because you can specify the exact rules you want to suppress following `scalafix:ok`. If it is followed by nothing then all rules are suppressed (wildcard). If you specify something, scalafix will parse it and disable the rules that there is a match.

At the moment you can add descriptions as long as you use the following format:

```
URLEncoder.encode(s, "UTF-8") /* scalafix:ok */ // UTF-8 always succeeds
```

**Comment by fommil** [comment]
I wonder if perhaps some marker, like another `//`, or a semicolon should allow arbitrary comments

**Comment by marcelocenerine** [comment]
yeah, could be

**Comment by olafurpg** [comment]
I think this is a good idea, do you have any suggestion how this can be supported alongside the syntax to target a single rule? For example, `replace(",.*", "")` could support something like

```scala
var x: String = null // scalafix:ok DisableSyntax.var, I need mutability
var x: String = null // scalafix:ok, I need mutability
```

**Comment by fommil** [comment]
comma is confusing because it is overloaded with the meaning in the `scalafix:off` syntax

**Comment by olafurpg** [comment]
do you have any suggestion how this can be supported alongside the syntax to target a single rule?

**Comment by fommil** [comment]
colon or semicolon instead of comma would be good

**Comment by olafurpg** [comment]
Can you give an example how that would look like?

**Comment by fommil** [comment]
```
 URLEncoder.encode(s, "UTF-8") // scalafix:ok; UTF-8 always succeeds
```

**Comment by marcelocenerine** [comment]
closed via #733
